PCIe 6.0 Packet Generator and Exerciser/PCIe 6.0 Packet Generator and Exerciser/Endpoint/memory_map.cpp

```diff
@@ -0,0 +1,164 @@
+#include "memory_map.h"
+
+MemoryMap::MemoryMap(uint64_t hostMemorySize, uint64_t deviceMemorySize)
+    : hostMemorySize_(hostMemorySize), deviceMemorySize_(deviceMemorySize)
+{
+    // Resize host memory to specified size
+    hostMemory_.resize(hostMemorySize_ / 4);
+
+    // Resize device memory to specified size
+    deviceMemory_.resize(deviceMemorySize_ / 4);
+}
+
+bool MemoryMap::read(uint64_t address, uint32_t* data)
+{
+    // Check if address is in host memory range
+    if (address < hostMemorySize_) {
+        *data = hostMemory_[address / 4];
+        return true;
+    }
+    // Check if address is in device memory range
+    else if (address >= 0x80000000 && address < (0x80000000 + deviceMemorySize_)) {
+        *data = deviceMemory_[(address - 0x80000000) / 4];
+        return true;
+    }
+    // Address is out of range
+    else {
+        return false;
```

Comment:
I can't understand how you know that the address is out of range without accessing the BARs.

Reply:
still to be implemented in another PR